src/pages/SignUp.jsx

```diff
@@ -1,19 +1,91 @@
 import React from 'react';
-import { Link } from 'react-router-dom';
+import {Link, useNavigate} from 'react-router-dom';
+import {useForm} from "react-hook-form";
+import axios from "axios";
 
 function SignUp() {
-  return (
-    <>
-      <h1>Registreren</h1>
-      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aspernatur atque consectetur, dolore eaque eligendi
-        harum, numquam, placeat quisquam repellat rerum suscipit ullam vitae. A ab ad assumenda, consequuntur deserunt
-        doloremque ea eveniet facere fuga illum in numquam quia reiciendis rem sequi tenetur veniam?</p>
-      <form>
-        <p>*Invoervelden*</p>
-      </form>
-      <p>Heb je al een account? Je kunt je <Link to="/signin">hier</Link> inloggen.</p>
-    </>
-  );
+    const {handleSubmit, formState: {errors}, register} = useForm();
+    const navigate = useNavigate();
+
+
+    async function handleFormSubmit(data) {
+        try {
+            const response = await axios.post(
+                "http://localhost:3000/register",
+                {
+                    email: data.email,
+                    password: data.password,
+                    username: data.username,
+                },
+                {
+                    headers: {
+                        "Content-Type": "application/json",
+                    },
+                }
+            );
+            console.log("Succesvol geregistreerd");
+            navigate("/signin")
+        } catch (e) {
+            if (e.response) {
```

Comment:
Zou dit niet een fantastische helper-functie zijn? Die kijkt welke error of error-code er opgeworpen wordt en een passende foutmelding teruggeeft? 😍